### DISCUSSION TITLE
Unity 2023, the TextMesh Pro package has been deprecated

### DISCUSSION THREAD

**Discussion by MargaritaChebotareva** [simonoliver/UnityFigmaBridge#41] [Q&A]
Hello! Thanks for your plugin!

In short, the problem is that in Unity 2023.2.3 the TMP library has been moved to com.unity.ugui and you can only get the essential resources for TMP version 2.0.1.
![unity_2023_2_3f1_need_TMP](https://github.com/simonoliver/UnityFigmaBridge/assets/43342457/d33ae3be-e240-4dfe-bb2f-5cb61e507a98)

I tried changing the TMP version in dependencies in package.json of your plugin or changed "com.unity.textmeshpro": "2.0.1" to "com.unity.ugui": "2.0.0" but it didn't work. 

In the end I have only version 2.0.1. In this case, using the plugin gives the following result:

![unity_2023_2_3f1_result](https://github.com/simonoliver/UnityFigmaBridge/assets/43342457/9bef5a21-ed42-41d6-88d7-825eeb988e88)

I downloaded Unity 2021.3.19f and ran your project. In the old version of Unity everything works, it looks like this:

![unity_2021_3_19f1_it_works](https://github.com/simonoliver/UnityFigmaBridge/assets/43342457/e262d7db-e0b5-42d7-bff1-36a916e7c006)

In the package manager, the TMP library is version 3.0.6, but as I said, I can't get it in the new version of Unity.

I tried to find a solution and found this thread:
https://forum.unity.com/threads/2023-2-latest-development-on-textmesh-pro.1434757/

Now, I'm thinking of taking the essential resources of TMP from your project and moving to 2023, but this is certainly not the best solution...

I think that if  change the library that is used in the code (I don't understand why it doesn't throw a compiler error due to namespaces), the plugin will use ugui's new version of TMP, but maybe I'm confusing something.

Could you please help me?

**Comment by simonoliver** [comment]
Hey thanks for letting me know about this - I'll make sure to take a look next time I work on this.
